src/Raven.Server/Integrations/PostgreSQL/PgSession.cs

```diff
@@ -0,0 +1,232 @@
+
+            if (database == null)
+            {
+                await writer.WriteAsync(messageBuilder.ErrorResponse(
+                    PgSeverity.Fatal,
+                    PgErrorCodes.ConnectionFailure,
+                    "Failed to connect to database",
+                    $"Database '{databaseName}' does not exist"), _token);
+                return;
+            }
+
+            try
+            {
+                var username = _clientOptions["user"];
+                using var transaction = new PgTransaction(database, new MessageReader(), username);
+
+                if (_serverCertificate != null)
+                {
+                    // Authentication is required only when running in secured mode
+
+                    await writer.WriteAsync(messageBuilder.AuthenticationCleartextPassword(), _token);
+                    var authMessage = await transaction.MessageReader.GetUninitializedMessage(reader, _token);
+                    await authMessage.Init(transaction.MessageReader, reader, _token);
+                    await authMessage.Handle(transaction, messageBuilder, reader, writer, _token);
+                }
+                else
+                {
+                    await writer.WriteAsync(messageBuilder.AuthenticationOk(), _token);
+                }
+
+                await writer.WriteAsync(messageBuilder.ParameterStatusMessages(PgConfig.ParameterStatusList), _token);
+                await writer.WriteAsync(messageBuilder.BackendKeyData(_processId, _identifier), _token);
+                await writer.WriteAsync(messageBuilder.ReadyForQuery(transaction.State), _token);
+
+                while (_token.IsCancellationRequested == false)
+                {
+                    var message = await transaction.MessageReader.GetUninitializedMessage(reader, _token);
+
+                    try
+                    {
+                        await message.Init(transaction.MessageReader, reader, _token);
+                        await message.Handle(transaction, messageBuilder, reader, writer, _token);
+                    }
+                    catch (PgErrorException e)
+                    {
+                        await message.HandleError(e, transaction, messageBuilder, writer, _token);
+                    }
+                }
+            }
+            catch (PgFatalException e)
+            {
+                if (Logger.IsInfoEnabled)
+                    Logger.Info($"{e.Message} (fatal pg error code {e.ErrorCode})", e);
+
+                await writer.WriteAsync(messageBuilder.ErrorResponse(
+                    PgSeverity.Fatal,
+                    e.ErrorCode,
+                    e.Message,
+                    e.ToString()), _token);
+            }
+            catch (PgErrorException e)
+            {
+                if (Logger.IsInfoEnabled)
+                    Logger.Info($"{e.Message} (pg error code {e.ErrorCode})", e);
```

Comment:
Provide details on the source connection?

Reply:
What kind of details would to expect to see here?

Reply:
IP, if certificate is used, its details?

Reply:
Sorry, this is PGSQL - so IP & Username, I think

Reply:
Fixed in https://github.com/ravendb/ravendb/pull/12989/commits/4e7574a25f9bc3cad16382c3feebce0eb83ab916